samples/cpp/benchmark_app/main.cpp

```diff
@@ -35,6 +35,41 @@
 #include "remote_tensors_filling.hpp"
 #include "statistics_report.hpp"
 #include "utils.hpp"
+
+#if defined _WIN32
+
+#include <windows.h>
+#include <psapi.h>
+
+int64_t getPeakMemoryUsage() {
+    PROCESS_MEMORY_COUNTERS memCounters;
+    GetProcessMemoryInfo(GetCurrentProcess(), &memCounters, sizeof(memCounters));
+    return memCounters.PeakWorkingSetSize / 1000;
+}
+
+#else
+
+#include <fstream>
+#include <regex>
+#include <sstream>
+
+int64_t getPeakMemoryUsage() {
+    size_t peakMemUsageKB = 0;
+
+    std::ifstream statusFile("/proc/self/status");
+    std::string line;
+    std::regex vmPeakRegex("VmPeak:");
+    std::smatch vmMatch;
+    while (std::getline(statusFile, line)) {
+        if (std::regex_search(line, vmMatch, vmPeakRegex)) {
+            std::istringstream iss(vmMatch.suffix());
+            iss >> peakMemUsageKB;
+        }
+    }
+    return static_cast<int64_t>(peakMemUsageKB);
```

Comment:
That didn't occur to me instantly why the return value is `int64_t` even though the memory usage is never negative. I think the reason is that you compute a difference later. Can you add a comment here so the return values doesn't get "fixed" later.

Reply:
added comment